pkg/apis/autoscaling/validation/validation.go

```diff
@@ -156,6 +159,65 @@ func validateMetrics(metrics []autoscaling.MetricSpec, fldPath *field.Path, minR
 	return allErrs
 }
 
+func validateBehavior(behavior *autoscaling.HorizontalPodAutoscalerBehavior, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if behavior != nil {
+		if scaleUpErrs := validateScalingRules(behavior.ScaleUp, fldPath.Child("scaleUp")); len(scaleUpErrs) > 0 {
+			allErrs = append(allErrs, scaleUpErrs...)
+		}
+		if scaleDownErrs := validateScalingRules(behavior.ScaleDown, fldPath.Child("scaleDown")); len(scaleDownErrs) > 0 {
+			allErrs = append(allErrs, scaleDownErrs...)
+		}
+	}
+	return allErrs
+}
+
+func validateScalingRules(rules *autoscaling.HPAScalingRules, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if rules != nil {
+		if rules.StabilizationWindowSeconds != nil && *rules.StabilizationWindowSeconds < 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("stabilizationWindowSeconds"), rules.StabilizationWindowSeconds, "must be greater than or equal to zero"))
+		}
+		if rules.StabilizationWindowSeconds != nil && *rules.StabilizationWindowSeconds > 3600 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("stabilizationWindowSeconds"), rules.StabilizationWindowSeconds, "must be less than or equal to 3600"))
+		}
+		if rules.SelectPolicy != nil &&
+			*rules.SelectPolicy != autoscaling.MaxPolicySelect &&
+			*rules.SelectPolicy != autoscaling.MinPolicySelect &&
+			*rules.SelectPolicy != autoscaling.DisabledPolicySelect {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("selectPolicy"), rules.SelectPolicy, "must be max, min, or disabled"))
```

Comment:
`field.NotSupported` is what you want here, and you can pass it a slice of valid values.